src/core/contracts/registry.ts

```diff
@@ -102,6 +107,36 @@ export const getDSNPRegistryUpdateEvents = async (
   });
 };
 
+/**
+ * validates a message against a signature and then checks that the signer has the permissions
+ * specified.
+ * @param signature the signature for the message
+ * @param message the DSNPMessage
+ * @param dsnpId the DSNP Id of the supposed signer
+ * @param permission the permissions to check for
+ * @param blockNumber (optional) the block number at which to perform this check.  Defaults to "latest"
+ */
+export const validateMessage = async (
+  signature: HexString,
+  message: DSNPMessage,
+  dsnpId: HexString,
+  permission: Permission,
+  blockNumber?: number
+): Promise<boolean> => {
+  const reg = await resolveId(dsnpId);
+  if (!reg) throw MissingContract;
+
+  const { provider } = getConfig();
+  if (!provider) throw MissingProvider;
+
+  const realBlockNumber = blockNumber ? blockNumber : "latest";
+  const bh = (await provider?.getBlock(realBlockNumber))?.number;
+  if (!bh) throw new Error("could not get block at height " + realBlockNumber);
```

Comment:
Do we need to do this test?

isAuthorizedTo can take future block numbers and the "forever" version is with block="0x0" which would be a fine default.

Reply:
Ah ok I didn't know that.